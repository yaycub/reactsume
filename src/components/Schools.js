import React from 'react';
import PropTypes from 'prop-types';

const Schools = ({ school }) => {
  const schoolElements = school.map((item, i) => {
    return (
      <li key={i}>
        <h4>{item.name}</h4>
        <p>{item.cert}</p>
      </li>
    );
  });

  return (
    <ul>
      <h3>Academic Background</h3>
      {schoolElements}
    </ul>
  );
};

Schools.propTypes = {
  school: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    cert: PropTypes.string.isRequired
  })).isRequired
};

export default Schools;
