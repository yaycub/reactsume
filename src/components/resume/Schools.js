import React from 'react';
import PropTypes from 'prop-types';
import styles from './styles/Schools.css';

const Schools = ({ school }) => {
  const schoolElements = school.map((item, i) => {
    return (
      <li key={i}>
        <h4>{item.name}</h4>
        <p>{item.cert}</p>
      </li>
    );
  });

  return (
    <ul className={styles.Schools}>
      <h3>Academic Background</h3>
      {schoolElements}
    </ul>
  );
};

Schools.propTypes = {
  school: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    cert: PropTypes.string.isRequired
  })).isRequired
};

export default Schools;
