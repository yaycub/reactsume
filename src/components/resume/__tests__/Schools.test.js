import React from 'react';
import { shallow } from 'enzyme';
import data from '../../../resume.json';
import Schools from '../Schools';

describe('Schools Component', () => {
  it('can render Schools', () => {
    const wrapper = shallow(<Schools {...data} />);
    expect(wrapper).toMatchSnapshot();
  });
});
