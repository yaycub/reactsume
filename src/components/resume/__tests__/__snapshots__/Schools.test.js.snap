// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schools Component can render Schools 1`] = `
<ul>
  <h3>
    Academic Background
  </h3>
  <li
    key="0"
  >
    <h4>
      Alchemy Code Lab
    </h4>
    <p>
      Full-Stack Developer
    </p>
  </li>
  <li
    key="1"
  >
    <h4>
      Hillsborough Community College
    </h4>
    <p>
      AA Engineering
    </p>
  </li>
</ul>
`;
